meshers/blocky/voxel.cpp

```diff
@@ -316,6 +370,23 @@ static void bake_mesh_geometry(Voxel &config, Voxel::BakedData &baked_data) {
 			tri_positions[1] = positions_read[indices_read[i + 1]];
 			tri_positions[2] = positions_read[indices_read[i + 2]];
 
+			float tangent[4];
+
+			if(tangents_empty){
+				//If tangents are empty then we calculate them
```

Comment:
I think tangents should be optional. Users should be allowed to provide a mesh without tangents if they don't need fancy textures. If they are missing and wanted, it's should be Godot importer/DCC exporter job to fix it.

Ideally, the presence of tangents should be determined per surface, if the material uses normalmaps. Or it would be a boolean on the library, but I'm not sure about the design of this (it's late here).

Reply:
I agree. The issue is that we'd need some sort of flag to toggle the mesh array in voxel_mesher_blocky.cpp because the visual server checks the packed array to be the exact correct size. Maybe I add a class flag for whether they're present?

Reply:
So I started to implement optional tangents based on whether they're included in the original mesh and if not, just passing an empty array. A problem came to mind that if I do it in a naive manner like this, then if we have a chunk sent to the blocky_mesher that includes a mesh with the tangents, and one without, the visual server will throw a fit because the count doesn't line up. 

I'll think on it some more, but I've made the rest of the requested changes.

Reply:
Yeah that last problem is what made me unsure, either way it has to be known up front, hence the idea of a boolean option

Reply:
Yeah. I see two possibilities that keep the library instance hidden from the voxel.cpp.

- Expose the toggle in the editor under VoxelLibrary
- Expose the toggle in the editor under Voxel

In both cases we create a method in the voxel library for looping through and setting the flag on existing voxels, and add a few lines to the methods that set/add voxels to the library to check or set the flags in the voxels being added. 

Design-wise in the editor I'm leaning towards under VoxelLibrary because the change is a global change and including it under Voxel doesn't reflect that. We'd just add an option like "Bake Tangents" or something under Voxel Count. 

Do you have any other ideas/what would you prefer?